### PR TITLE
use configSTACK_DEPTH_TYPE consequently (updated for 11.0.x)

### PR DESCRIPTION
**Pull request by feilipu**
Title
-----------
Use `configSTACK_DEPTH_TYPE` consequently throughout.

Description
-----------
The configuration `configSTACK_DEPTH_TYPE` is used partially, but many places the stack depth variable type is assumed to be `uint16_t`, and in others the assumption is `uint32_t`, and the variable names are sometimes `us...` or `ul...` respectively.

This PR implements consequent use of `configSTACK_DEPTH_TYPE`  throughout and adjusts variable names to be `ux...` or `pux...` as appropriate.

The [default value remains set as __`uint16_t`__](https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/main/include/FreeRTOS.h#L2759C47-L2765C47), although some ports assume `uint32_t` without necessarily redefining it as advised.

Files have been updated for updated for 11.0.x.

Note that some extended coverage tests fail because they don't use the correct variable size, as some ports don't configure the `configSTACK_DEPTH_TYPE` correctly. Rather they incorrectly assume the default is `uint32_t`.

Just as a side note; the contents of this PR have been tested by 1,000s of users over the past [6 years](https://github.com/aws/amazon-freertos/issues/12) in [Arduino_FreeRTOS](https://www.arduino.cc/reference/en/libraries/freertos/). Anyone who has come to learn FreeRTOS via the Arduino IDE will expect the stack type behaviour found in this PR.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: portable/IAR/ARM_CM55_NTZ/non_secure/port.c [hunk 1]
```diff
@@ -1775,7 +1775,7 @@ void vPortEndScheduler( void ) /* PRIVILEGED_FUNCTION */
     void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                     const struct xMEMORY_REGION * const xRegions,
                                     StackType_t * pxBottomOfStack,
-                                    uint32_t ulStackDepth )
+                                    vPortStoreTaskMPUSettings ulStackDepth )
```
Comment (kar-rahul-aws) [inline portable/IAR/ARM_CM55_NTZ/non_secure/port.c:1778]: This has to be replaced with configSTACK_DEPTH_TYPE

File: portable/Common/mpu_wrappers.c [hunk 2]
```diff
@@ -91,7 +91,7 @@
     #if ( configSUPPORT_STATIC_ALLOCATION == 1 )
         TaskHandle_t MPU_xTaskCreateStatic( TaskFunction_t pxTaskCode,
                                             const char * const pcName,
-                                            const uint32_t ulStackDepth,
+                                            configSTACK_DEPTH_TYPE uxStackDepth,
```
Comment (kar-rahul-aws) [inline portable/Common/mpu_wrappers.c:94]: ``const`` keyword has been removed, it was present in in the initial parameter list for uxStackDepth

File: portable/IAR/ARM_CM55_NTZ/non_secure/port.c (hunk 1 above)
Comment (feilipu) [reply portable/IAR/ARM_CM55_NTZ/non_secure/port.c]: Sharp eyes! Fixed.

File: portable/Common/mpu_wrappers.c (hunk 2 above)
Comment (feilipu) [reply portable/Common/mpu_wrappers.c]: Fixed.

Also adjusted `MPU_xTaskCreate()` so that it follows its prototypes too.

**Comment by Skptak** [comment]
In the commit that introduced `configSTACK_DEPTH_TYPE` ([link](https://github.com/FreeRTOS/FreeRTOS-Kernel/commit/225f13bac2cdf08668faadf510bb9b37fe2bbc38)), it's mentioned that reason this was introduced was to keep the existing APIs backwards compatible.

Changing the variable name from `ulStackDepth` to `uxStackDepth`, while correct according to the [FreeRTOS Naming Convention](https://www.freertos.org/FreeRTOS-Coding-Standard-and-Style-Guide.html), would break backwards compatibility

In my opinion, if we're going to introduce a backwards compatibility breaking change like this, we should instead remove  `configSTACK_DEPTH_TYPE` and use `StackType_t` instead. This way the value would then be the correct bit width for any architecture it is running on, instead of relying upon a separate configuration option?

**Comment by feilipu** [comment]
> We should instead remove `configSTACK_DEPTH_TYPE` and use `StackType_t` instead.

That path fails for 8-bit devices, often (usually) need more than 256 bytes of stack for each Task. Hence the configuration default has always been set to `uint16_t`.

Unfortunately also, historically many ports didn't change that default setting to something sensible or efficient for their architecture. So there is mixed variable casting between 16-bit and 32-bit code, which has been addressed in this PR.

And the code as it stands today it is inefficient on 64-bit architectures, which has also been addressed by this PR too.

The new `FreeRTOSConfig.h` template file provides `StackType_t` as the configuration for `configSTACK_DEPTH_TYPE` (rather than the default value). Perhaps this is the best default going forward, as you have suggested?

But the stack depth type always needs to remain configurable, to cover all device architectures.

**Comment by feilipu** [comment]
Just as a side note; the contents of this PR have been tested by 1,000s of users over the past [6 years](https://github.com/aws/amazon-freertos/issues/12) in [Arduino_FreeRTOS](https://www.arduino.cc/reference/en/libraries/freertos/). So anyone who has come to learn FreeRTOS via Arduino will expect the stack type behaviour found in this PR.

File: portable/GCC/ARM_CM23/non_secure/port.c [hunk 3]
```diff
@@ -1775,7 +1775,7 @@ void vPortEndScheduler( void ) /* PRIVILEGED_FUNCTION */
     void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                     const struct xMEMORY_REGION * const xRegions,
                                     StackType_t * pxBottomOfStack,
-                                    uint32_t ulStackDepth )
+                                    configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/GCC/ARM_CM23/non_secure/port.c:1778]:
```suggestion
                                    configSTACK_DEPTH_TYPE uxStackDepth )
```

File: portable/GCC/ARM_CM23_NTZ/non_secure/port.c [hunk 4]
```diff
@@ -1775,7 +1775,7 @@ void vPortEndScheduler( void ) /* PRIVILEGED_FUNCTION */
     void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                     const struct xMEMORY_REGION * const xRegions,
                                     StackType_t * pxBottomOfStack,
-                                    uint32_t ulStackDepth )
+                                    configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/GCC/ARM_CM23_NTZ/non_secure/port.c:1778]:
```suggestion
                                    configSTACK_DEPTH_TYPE uxStackDepth )
```

File: portable/GCC/ARM_CM33/non_secure/port.c [hunk 5]
```diff
@@ -1775,7 +1775,7 @@ void vPortEndScheduler( void ) /* PRIVILEGED_FUNCTION */
     void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                     const struct xMEMORY_REGION * const xRegions,
                                     StackType_t * pxBottomOfStack,
-                                    uint32_t ulStackDepth )
+                                    configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/GCC/ARM_CM33/non_secure/port.c:1778]:
```suggestion
                                    configSTACK_DEPTH_TYPE uxStackDepth )
```

File: portable/GCC/ARM_CM33_NTZ/non_secure/port.c [hunk 6]
```diff
@@ -1775,7 +1775,7 @@ void vPortEndScheduler( void ) /* PRIVILEGED_FUNCTION */
     void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                     const struct xMEMORY_REGION * const xRegions,
                                     StackType_t * pxBottomOfStack,
-                                    uint32_t ulStackDepth )
+                                    configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/GCC/ARM_CM33_NTZ/non_secure/port.c:1778]:
```suggestion
                                    configSTACK_DEPTH_TYPE uxStackDepth )
```

**Review by n9wxu — DISMISSED** [review]

File: portable/GCC/ARM_CM35P/non_secure/port.c [hunk 7]
```diff
@@ -1775,7 +1775,7 @@ void vPortEndScheduler( void ) /* PRIVILEGED_FUNCTION */
     void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                     const struct xMEMORY_REGION * const xRegions,
                                     StackType_t * pxBottomOfStack,
-                                    uint32_t ulStackDepth )
+                                    configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/GCC/ARM_CM35P/non_secure/port.c:1778]:
```suggestion
                                    configSTACK_DEPTH_TYPE uxStackDepth )
```

File: portable/GCC/ARM_CM35P_NTZ/non_secure/port.c [hunk 8]
```diff
@@ -1775,7 +1775,7 @@ void vPortEndScheduler( void ) /* PRIVILEGED_FUNCTION */
     void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                     const struct xMEMORY_REGION * const xRegions,
                                     StackType_t * pxBottomOfStack,
-                                    uint32_t ulStackDepth )
+                                    configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/GCC/ARM_CM35P_NTZ/non_secure/port.c:1778]:
```suggestion
                                    configSTACK_DEPTH_TYPE uxStackDepth )
```

File: portable/GCC/ARM_CM3_MPU/port.c [hunk 9]
```diff
@@ -1227,7 +1227,7 @@ void vPortSwitchToUserMode( void )
 void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                 const struct xMEMORY_REGION * const xRegions,
                                 StackType_t * pxBottomOfStack,
-                                uint32_t ulStackDepth )
+                                configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/GCC/ARM_CM3_MPU/port.c:1230]:
```suggestion
                                configSTACK_DEPTH_TYPE uxStackDepth )
```

File: portable/GCC/ARM_CM4_MPU/port.c [hunk 10]
```diff
@@ -1385,7 +1385,7 @@ void vPortSwitchToUserMode( void )
 void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                 const struct xMEMORY_REGION * const xRegions,
                                 StackType_t * pxBottomOfStack,
-                                uint32_t ulStackDepth )
+                                configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/GCC/ARM_CM4_MPU/port.c:1388]:
```suggestion
                                configSTACK_DEPTH_TYPE uxStackDepth )
```

File: portable/GCC/ARM_CM55/non_secure/port.c [hunk 11]
```diff
@@ -1775,7 +1775,7 @@ void vPortEndScheduler( void ) /* PRIVILEGED_FUNCTION */
     void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                     const struct xMEMORY_REGION * const xRegions,
                                     StackType_t * pxBottomOfStack,
-                                    uint32_t ulStackDepth )
+                                    configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/GCC/ARM_CM55/non_secure/port.c:1778]:
```suggestion
                                    configSTACK_DEPTH_TYPE uxStackDepth )
```

File: portable/GCC/ARM_CM55_NTZ/non_secure/port.c [hunk 12]
```diff
@@ -1775,7 +1775,7 @@ void vPortEndScheduler( void ) /* PRIVILEGED_FUNCTION */
     void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                     const struct xMEMORY_REGION * const xRegions,
                                     StackType_t * pxBottomOfStack,
-                                    uint32_t ulStackDepth )
+                                    configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/GCC/ARM_CM55_NTZ/non_secure/port.c:1778]:
```suggestion
                                    configSTACK_DEPTH_TYPE uxStackDepth )
```

File: portable/GCC/ARM_CM85/non_secure/port.c [hunk 13]
```diff
@@ -1775,7 +1775,7 @@ void vPortEndScheduler( void ) /* PRIVILEGED_FUNCTION */
     void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                     const struct xMEMORY_REGION * const xRegions,
                                     StackType_t * pxBottomOfStack,
-                                    uint32_t ulStackDepth )
+                                    configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/GCC/ARM_CM85/non_secure/port.c:1778]:
```suggestion
                                    configSTACK_DEPTH_TYPE uxStackDepth )
```

File: portable/GCC/ARM_CM85_NTZ/non_secure/port.c [hunk 14]
```diff
@@ -1775,7 +1775,7 @@ void vPortEndScheduler( void ) /* PRIVILEGED_FUNCTION */
     void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                     const struct xMEMORY_REGION * const xRegions,
                                     StackType_t * pxBottomOfStack,
-                                    uint32_t ulStackDepth )
+                                    configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/GCC/ARM_CM85_NTZ/non_secure/port.c:1778]:
```suggestion
                                    configSTACK_DEPTH_TYPE uxStackDepth )
```

File: portable/IAR/ARM_CM23/non_secure/port.c [hunk 15]
```diff
@@ -1775,7 +1775,7 @@ void vPortEndScheduler( void ) /* PRIVILEGED_FUNCTION */
     void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                     const struct xMEMORY_REGION * const xRegions,
                                     StackType_t * pxBottomOfStack,
-                                    uint32_t ulStackDepth )
+                                    configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/IAR/ARM_CM23/non_secure/port.c:1778]:
```suggestion
                                    configSTACK_DEPTH_TYPE uxStackDepth )
```

File: portable/IAR/ARM_CM23_NTZ/non_secure/port.c [hunk 16]
```diff
@@ -1775,7 +1775,7 @@ void vPortEndScheduler( void ) /* PRIVILEGED_FUNCTION */
     void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                     const struct xMEMORY_REGION * const xRegions,
                                     StackType_t * pxBottomOfStack,
-                                    uint32_t ulStackDepth )
+                                    configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/IAR/ARM_CM23_NTZ/non_secure/port.c:1778]:
```suggestion
                                    configSTACK_DEPTH_TYPE uxStackDepth )
```

File: portable/IAR/ARM_CM33/non_secure/port.c [hunk 17]
```diff
@@ -1775,7 +1775,7 @@ void vPortEndScheduler( void ) /* PRIVILEGED_FUNCTION */
     void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                     const struct xMEMORY_REGION * const xRegions,
                                     StackType_t * pxBottomOfStack,
-                                    uint32_t ulStackDepth )
+                                    configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/IAR/ARM_CM33/non_secure/port.c:1778]:
```suggestion
                                    configSTACK_DEPTH_TYPE uxStackDepth )
```

File: portable/IAR/ARM_CM33_NTZ/non_secure/port.c [hunk 18]
```diff
@@ -1775,7 +1775,7 @@ void vPortEndScheduler( void ) /* PRIVILEGED_FUNCTION */
     void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                     const struct xMEMORY_REGION * const xRegions,
                                     StackType_t * pxBottomOfStack,
-                                    uint32_t ulStackDepth )
+                                    configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/IAR/ARM_CM33_NTZ/non_secure/port.c:1778]:
```suggestion
                                    configSTACK_DEPTH_TYPE uxStackDepth )
```

File: portable/IAR/ARM_CM35P/non_secure/port.c [hunk 19]
```diff
@@ -1775,7 +1775,7 @@ void vPortEndScheduler( void ) /* PRIVILEGED_FUNCTION */
     void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                     const struct xMEMORY_REGION * const xRegions,
                                     StackType_t * pxBottomOfStack,
-                                    uint32_t ulStackDepth )
+                                    configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/IAR/ARM_CM35P/non_secure/port.c:1778]:
```suggestion
                                    configSTACK_DEPTH_TYPE uxStackDepth )
```

File: portable/IAR/ARM_CM35P_NTZ/non_secure/port.c [hunk 20]
```diff
@@ -1775,7 +1775,7 @@ void vPortEndScheduler( void ) /* PRIVILEGED_FUNCTION */
     void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                     const struct xMEMORY_REGION * const xRegions,
                                     StackType_t * pxBottomOfStack,
-                                    uint32_t ulStackDepth )
+                                    configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/IAR/ARM_CM35P_NTZ/non_secure/port.c:1778]:
```suggestion
                                    configSTACK_DEPTH_TYPE uxStackDepth )
```

File: portable/IAR/ARM_CM4F_MPU/port.c [hunk 21]
```diff
@@ -1127,7 +1127,7 @@ static uint32_t prvGetMPURegionSizeSetting( uint32_t ulActualSizeInBytes )
 void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                 const struct xMEMORY_REGION * const xRegions,
                                 StackType_t * pxBottomOfStack,
-                                uint32_t ulStackDepth )
+                                configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/IAR/ARM_CM4F_MPU/port.c:1130]:
```suggestion
                                    configSTACK_DEPTH_TYPE uxStackDepth )
```

File: portable/IAR/ARM_CM55/non_secure/port.c [hunk 22]
```diff
@@ -1775,7 +1775,7 @@ void vPortEndScheduler( void ) /* PRIVILEGED_FUNCTION */
     void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                     const struct xMEMORY_REGION * const xRegions,
                                     StackType_t * pxBottomOfStack,
-                                    uint32_t ulStackDepth )
+                                    configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/IAR/ARM_CM55/non_secure/port.c:1778]:
```suggestion
                                    configSTACK_DEPTH_TYPE uxStackDepth )
```

File: portable/IAR/ARM_CM55_NTZ/non_secure/port.c [hunk 23]
```diff
@@ -1775,7 +1775,7 @@ void vPortEndScheduler( void ) /* PRIVILEGED_FUNCTION */
     void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                     const struct xMEMORY_REGION * const xRegions,
                                     StackType_t * pxBottomOfStack,
-                                    uint32_t ulStackDepth )
+                                    configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/IAR/ARM_CM55_NTZ/non_secure/port.c:1778]:
```suggestion
                                    configSTACK_DEPTH_TYPE uxStackDepth )
```

File: portable/IAR/ARM_CM85/non_secure/port.c [hunk 24]
```diff
@@ -1775,7 +1775,7 @@ void vPortEndScheduler( void ) /* PRIVILEGED_FUNCTION */
     void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                     const struct xMEMORY_REGION * const xRegions,
                                     StackType_t * pxBottomOfStack,
-                                    uint32_t ulStackDepth )
+                                    configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/IAR/ARM_CM85/non_secure/port.c:1778]:
```suggestion
                                    configSTACK_DEPTH_TYPE uxStackDepth )
```

File: portable/IAR/ARM_CM85_NTZ/non_secure/port.c [hunk 25]
```diff
@@ -1775,7 +1775,7 @@ void vPortEndScheduler( void ) /* PRIVILEGED_FUNCTION */
     void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                     const struct xMEMORY_REGION * const xRegions,
                                     StackType_t * pxBottomOfStack,
-                                    uint32_t ulStackDepth )
+                                    configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/IAR/ARM_CM85_NTZ/non_secure/port.c:1778]:
```suggestion
                                    configSTACK_DEPTH_TYPE uxStackDepth )
```

File: portable/RVDS/ARM_CM4_MPU/port.c [hunk 26]
```diff
@@ -1380,7 +1380,7 @@ void vPortSwitchToUserMode( void )
 void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                 const struct xMEMORY_REGION * const xRegions,
                                 StackType_t * pxBottomOfStack,
-                                uint32_t ulStackDepth )
+                                configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/RVDS/ARM_CM4_MPU/port.c:1383]:
```suggestion
                                    configSTACK_DEPTH_TYPE uxStackDepth )
```

**Comment by n9wxu** [comment]
I like these changes and have approved them.  We need a matching PR to fix the tests and demo's.  Briefly looking at these CI failures shows there are only a few places where changes are required.

File: portable/ARMv8M/non_secure/port.c [hunk 27]
```diff
@@ -1775,7 +1775,7 @@ void vPortEndScheduler( void ) /* PRIVILEGED_FUNCTION */
     void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                     const struct xMEMORY_REGION * const xRegions,
                                     StackType_t * pxBottomOfStack,
-                                    uint32_t ulStackDepth )
+                                    configSTACK_DEPTH_TYPE ulStackDepth )
     {
```
Comment (Skptak) [inline portable/ARMv8M/non_secure/port.c:1779]:
```suggestion
                                    configSTACK_DEPTH_TYPE uxStackDepth )
```

File: portable/ARMv8M/non_secure/port.c [hunk 28]
```diff
@@ -1775,7 +1775,7 @@ void vPortEndScheduler( void ) /* PRIVILEGED_FUNCTION */
     void vPortStoreTaskMPUSettings( xMPU_SETTINGS * xMPUSettings,
                                     const struct xMEMORY_REGION * const xRegions,
                                     StackType_t * pxBottomOfStack,
-                                    uint32_t ulStackDepth )
+                                    configSTACK_DEPTH_TYPE ulStackDepth )
```
Comment (Skptak) [inline portable/ARMv8M/non_secure/port.c:1778]:
```suggestion
                                    configSTACK_DEPTH_TYPE uxStackDepth )
```

**Review by sirnish — APPROVED** [review]

**Comment by aggarg** [comment]
I have made the following changes in the recent commit:
1. Default `configSTACK_DEPTH_TYPE` to `StackType_t`.
2. Do not rename `TaskParameters_t.usStackDepth` and `TaskStatus_t.usStackHighWaterMark` as this would avoid breaking existing applications. More than the existing applications, I am concerned about any existing FreeRTOS aware debug tools which might be using `TaskStatus_t`.

Other than those there were some build failures which I have addressed. Let me know if you are good with these and we will merge this PR.

**Review by aggarg — APPROVED** [review]

**Comment by feilipu** [comment]
> Default `configSTACK_DEPTH_TYPE` to `StackType_t`.

Agreed that is the best solution [as discussed](https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/942#issuecomment-1879370945) above.

Perhaps (in a separate PR) some 8 bit CPU ports will need to have their [`FreeRTOSConfig.h`](https://www.freertos.org/a00110.html) configuration set to `uint16_t`, if it is not already the case.

Also the documentation will need to be updated in [certain places](https://www.freertos.org/a00110.html) to reflect the new default.

> Do not rename `TaskParameters_t.usStackDepth` and `TaskStatus_t.usStackHighWaterMark` as this would avoid breaking existing applications. More than the existing applications, I am concerned about any existing FreeRTOS aware debug tools which might be using `TaskStatus_t`.

Agreed. Noting that this path is not compliant with the FreeRTOS variable naming standards, which may lead to later confusion.

> Other than those there were some build failures which I have addressed.

Thanks.

> Let me know if you are good with these and we will merge this PR.

Please proceed.

**Review by kar-rahul-aws — APPROVED** [review]

**Comment by kar-rahul-aws** [comment]
Thanks @feilipu . We will update the documentation for the changes.

**Pull request merged**